EIPS/eip-2253.md

```diff
@@ -0,0 +1,172 @@
+---
+eip: 2253
+title: wallet_getAddressBook JSON-RPC Method
+author: Loredana Cirstea (@loredanacirstea)
+discussions-to: https://ethereum-magicians.org/t/eip-2253-add-wallet-getaddressbook-json-rpc-method/3592
+status: Draft
+type: Standards Track
+category: ERC
+created: 2019-08-27
+requires: 1474
+---
+
+## Simple Summary
+
+This is a proposal for a new JSON-RPC call for retrieving a selection of items from a wallet's address book, with the user's permission.
+
+## Abstract
+
+There is no standardized way for a dApp to request address book information from a user's wallet. This leads to either nonoptimal UX, where the user needs to find and copy-paste an Ethereum address, or to duplication of effort across dApps and decreased privacy.
+
+## Motivation
+
+Address books for wallets are becoming a standard feature, greatly improving usability, while maintaining user ownership on this data.
+
+There is no standardized way for a dApp to leverage this information, with the user's permission. dApps can use the `wallet_getAddressBook` JSON-RPC method to retrieve strictly the information that they need from a wallet, in order to process the request.
+
+The current proposal will avoid duplication of effort - dApps will not need to ask the user to create another address book with them. And it also protects the user's privacy, while increasing usability.
+
+## Specification
+
+New JSON-RPC method: `wallet_getAddressBook`.
+
+**Arguments:**
+- filter object, optional:
+  - `count` - type `uint256`, the number of address book records requested; optional - when missing, the entire address book is requested
```

Comment:
I'm not sure about this. If there is a `count` at all, then it needs (?) to be an offset as well right? Otherwise you get this weird case where you can either get the (first?) ten, or the full thing, but not items `10` to `20`.

Reply:
I was thinking of `count` not as a filter with `limit`, `offset`, but as how many addresses are expected by the dApp. The user would control which addresses are selected and the wallet would control what/how/how many addresses are presented for selection (based on the user's data).

E.g. maybe the dApp only requires one such address - not the entire address book. Or 3 addresses - a dApp that says: you can send our asset for free to 3 friends! -> the wallet shows that you can select up to three addresses, so you can select 1, 2 or 3 addresses. (In [EIP-2256](https://github.com/ethereum/EIPs/pull/2256) I mention: `the user can amend the request by lowering the number of owned assets returned to the dApp`; I see I did not add this idea here)

This said, `count` clearly overlaps with the already known `filter` concepts. So, I am open to other names. Maybe:
- replace `filter object` with `options object`
- replace `count` with `limit` or `maxNumber`

Reply:
done:
s/filter/options/
s/count/limit/

I added the mention: 
"It remains to be discussed if the request parameters need to specify `optional` and `required` fields or minimum and maximum number of items to be returned."